AddOns/BasicFinancials/test/src/TestBasicBF.Codeunit.al

```diff
@@ -0,0 +1,301 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 139502 "Test Basic BF"
+{
+    Subtype = Test;
+    TestPermissions = Disabled;
+
+    var
+        Assert: Codeunit Assert;
+        AllProfileFilterTxt: Label 'MANUFACTURING|PROJECTS|SERVICES|WAREHOUSE|SHIPPING AND RECEIVING - WMS|SHIPPING AND RECEIVING|WAREHOUSE WORKER - WMS|PRODUCTION PLANNER|PROJECT MANAGER|DISPATCHER', Locked = true, Comment = 'As default the profile "SALES AND RELATIONSHIP MANAGER" cannot disable because it is set up as a default profile for one or more users or user groups.';
+        NotSupportedLicensesErr: Label 'Validation error for Field: IsSupportedLicenses,  Message = ''At least one user must have the Basic license.''', Locked = true;
+        NotSupportedCompaniesErr: Label 'Validation error for Field: IsSupportedCompanies,  Message = ''Exactly one company must exists in the environment.''', Locked = true;
+
+    trigger OnRun();
+    begin
+        // [FEATURE] [BASIC] [USER EXPERIENCE]
+    end;
+
+    [Test]
+    procedure TestBasicExperienceTierAfterInstall();
+    var
+    begin
+        // [SCENARIO] Check the Basic Experience Tier After Install of Basic Extension 
+        // [GIVEN] Application Area
+        // [WHEN]
+
+        // [THEN] Basic Experience Tier is Enabled
+        TestEnabledBasicExperienceTier();
+    end;
+
+    [Test]
+    procedure TestBasicApplicationAreaAfterInstall();
+    var
+    begin
+        // [SCENARIO] Check the Basic Application Area After Install of Basic Extension 
+        // [GIVEN] Application Area
+        // [WHEN]
+
+        // [THEN] Basic Application Area is Enabled
+        TestEnabledBasicApplicationArea();
+    end;
+
+    [Test]
+    procedure TestDisableRoleCenterAfterInstall();
+    var
+    begin
+        // [SCENARIO] Check disable Role Center After Install of Basic Extension 
+        // [GIVEN] Role Center
+        // [WHEN]
+
+        // [THEN] Role Center is disable
+        TestDisableRoleCenter();
+    end;
+
+    [Test]
+    procedure TestSetExperienceTierToEssential();
```

Comment:
Is there any value coming from this test ? As this is something more related to   ApplicationAreaMgmtFacade

Reply:
Yes, I think negative test is important, and provide value. The check that normal behavior still working and it is possible to set Experience Tier To Essential

Reply:
Well, I agree with you that the negative test is valuable. My point was that it should not belong to this extension as it tests only ApplicationAreaMgmtFacade. Will leave it to you.